k8snodedecorator/labels.go

```diff
@@ -0,0 +1,90 @@
+package k8snodedecorator
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"strconv"
+	"strings"
+
+	metadata "github.com/linode/go-metadata"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+func SetLabel(node *corev1.Node, key, newValue string) (changed bool) {
```

Comment:
optional: It doesn't look like this function is called from outside the `k8snodedecorator` package so it might make sense to make it private.